src/libraries/Enableable.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {RolesConsumer} from "src/modules/ROLES/OlympusRoles.sol";
+
+/// @title  Enableable
+/// @notice This contract is designed to be inherited by contracts that need to be enabled or disabled. It replaces the inconsistent usage of `active` and `locallyActive` state variables across the codebase.
+/// @dev    A contract that inherits from this contract should use the `whenEnabled` and `whenDisabled` modifiers to gate access to certain functions.
+///
+///         If custom logic and/or parameters are needed for the enable/disable functions, the inheriting contract can override the `_enable()` and `_disable()` functions. For example, `enable()` could be called with initialisation data that is decoded, validated and assigned in `_enable()`.
+abstract contract Enableable is RolesConsumer {
+    // ===== STATE VARIABLES ===== //
+
+    bool public isEnabled;
+    bytes32 public constant ROLE = "emergency_shutdown";
+
+    // ===== ERRORS ===== //
+
+    error NotDisabled();
+    error NotEnabled();
+
+    // ===== EVENTS ===== //
+
+    event Disabled();
+    event Enabled();
+
+    // ===== MODIFIERS ===== //
+
+    modifier whenEnabled() {
+        if (!isEnabled) revert NotEnabled();
+        _;
+    }
+
+    modifier whenDisabled() {
+        if (isEnabled) revert NotDisabled();
+        _;
+    }
+
+    // ===== ENABLEABLE FUNCTIONS ===== //
+
+    /// @notice Enable the contract
+    /// @dev    This function performs the following steps:
+    ///         1. Validates that the caller has `ROLE` ("emergency_shutdown")
+    ///         2. Validates that the contract is disabled
+    ///         3. Calls the implementation-specific `_enable()` function
+    ///         4. Changes the state of the contract to enabled
+    ///         5. Emits the `Enabled` event
+    ///
+    /// @param  enableData_ The data to pass to the implementation-specific `_enable()` function
+    function enable(bytes calldata enableData_) public onlyRole(ROLE) whenDisabled {
```

Comment:
Since this is can be pulled into all policies - I wonder if it'd be worth reducing bytecode by having a `toggleIsEnabled()` instead of distinct enable and disable functions?

The modifier could be also be a single:
```
    modifier whenEnabledIs(bool condition) {
        if (isEnabled == condition) revert IsEnabledNotMatching(condition);
        _;
    }
```

Tradeoff with being explicit - so don't mind, just throwing it out there (MonoCooler is kinda chunky which is why i bring up)

Reply:
I feel that the simplicity would be lost, and it also adds ambiguity.

e.g. the `whenEnabledIs(bool condition)` modifier you shared is reverting if `isEnabled == condition`, but I would have interpreted it the other way - `whenEnabledIs(true)` would revert if `isEnabled != true`

Reply:
Totally fair and agree, thought i'd just throw it in the mix.